cg/apps/demultiplex/sample_sheet/index.py

```diff
@@ -127,34 +142,75 @@ def pad_index_two(index_string: str, reverse_complement: bool) -> str:
-    (version 1.5).
-    """
-    LOG.info("Fix so that all indexes are in the correct format")
-    reverse_complement: bool = is_reverse_complement(run_parameters=run_parameters)
+    """Updates the barcode mismatches for both indexes of a FlowCellSampleNovaSeqX."""
+    index_1_sample_to_update, index_2_sample_to_update = get_index_pair(sample=sample_to_update)
     for sample in samples:
-        index1, index2 = sample.index.split("-")
-        index1: str = index1.strip()
-        index2: str = index2.strip()
-        index_length = len(index1)
-        if is_padding_needed(
-            index_cycles=run_parameters.index_length, sample_index_length=index_length
+        if sample_to_update.sample_id == sample.sample_id:
+            continue
+        index_1, index_2 = get_index_pair(sample=sample)
+        if (
+            get_hamming_distance_index_1(sequence_1=index_1_sample_to_update, sequence_2=index_1)
+            < MINIMUM_HAMMING_DISTANCE
+        ):
+            sample_to_update.barcode_mismatches_1 = 0
+        if (
+            get_hamming_distance_index_2(sequence_1=index_2_sample_to_update, sequence_2=index_2)
+            < MINIMUM_HAMMING_DISTANCE
         ):
-            LOG.debug("Padding indexes")
-            index1 = pad_index_one(index_string=index1)
-            index2 = pad_index_two(index_string=index2, reverse_complement=reverse_complement)
-        if reverse_complement:
-            index2 = get_reverse_complement_dna_seq(index2)
-        sample.index = index1
-        sample.index2 = index2
+            sample_to_update.barcode_mismatches_2 = 0
 
 
-def is_dual_index(index: str) -> bool:
-    """Determines if an index in the raw sample sheet is dual index or not."""
-    return "-" in index
+def adapt_indexes_for_sample(
+    sample: FlowCellSample, index_cycles: int, reverse_complement: bool
+) -> None:
+    """Adapts the indexes of sample.
+    1. Pad indexes if needed so that all indexes have a length equal to the number of index reads
+    2. Takes the reverse complement of index 2 in case of the new NovaSeq software control version
+    (1.7) in combination with the new reagent kit (version 1.5).
+    3. Assigns the indexes to the sample attributes index1 and index2."""
+    index1, index2 = get_index_pair(sample=sample)
+    index_length = len(index1)
+    if is_padding_needed(index_cycles=index_cycles, sample_index_length=index_length):
+        LOG.debug("Padding indexes")
+        index1 = pad_index_one(index_string=index1)
+        index2 = pad_index_two(index_string=index2, reverse_complement=reverse_complement)
+    if reverse_complement:
+        index2 = get_reverse_complement_dna_seq(index2)
+    sample.index = index1
+    sample.index2 = index2
+
+
+def adapt_samples(
+    samples: List[FlowCellSample],
+    run_parameters: RunParameters,
+) -> None:
+    """Adapt the samples with the correct index and barcode mismatch values."""
```

Comment:
What does adapt mean? Maybe the name of the function or the doc string can be made more descriptive